go/libraries/doltcore/rowconv/joiner.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rowconv
+
+import (
+	"errors"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/row"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type ColNamingFunc func(colName string) string
+
+type stringUint64Tuple struct {
+	str string
+	u64 uint64
+}
+
+type NamedSchema struct {
+	Name string
+	Sch  schema.Schema
+}
+
+type Joiner struct {
+	srcSchemas map[string]schema.Schema
+	tagMaps    map[string]map[uint64]uint64
+	revTagMap  map[uint64]stringUint64Tuple
+	tags       map[string][]uint64
+	joined     schema.Schema
+}
+
+func NewJoiner(namedSchemas []NamedSchema, namers map[string]ColNamingFunc) (*Joiner, error) {
```

Comment:
A comment about the high-level purpose of the joiner interface and what it does would be nice

Reply:
updated